capsules/src/ros.rs

```diff
@@ -0,0 +1,142 @@
+//! Read Only Systems calls
+//!
+//! This capsule provides read only system calls to userspace applications.
+//! This is similar to the Linux vDSO syscalls.
+//!
+//! The benefit of using these is that applications can avoid the context
+//! switch overhead of traditional syscalls by just reading the value from
+//! memory.
+//!
+//! The value will only be as accurate as the last time the application was
+//! switched to by the kernel.
+//!
+//! The layout of the read only syscalls in the allow region depends on the
+//! version. Userspace can use `command 0` to get the version information.
+//!
+//! Versions are backwards compatible, that is new versions will only add
+//! fields, not remove existing ones or change the order.
+//!
+//! Version 1:
+//!   |-------------------------|
+//!   |       Count (u32)       |
+//!   |-------------------------|
+//!   |   Pending Tasks (u32)   |
+//!   |-------------------------|
+//!   |                         |
+//!   |     Time Ticks (u64)    |
+//!   |-------------------------|
+//!
+
+use core::cell::Cell;
+use kernel::grant::Grant;
+use kernel::hil::time::{Ticks, Time};
+use kernel::platform::ContextSwitchOps;
+use kernel::process::{self, ProcessId};
+use kernel::processbuffer::{SharedProcessBuffer, WriteableProcessBuffer};
+use kernel::syscall::{CommandReturn, SyscallDriver};
+use kernel::ErrorCode;
+
+/// Syscall driver number.
+pub const DRIVER_NUM: usize = 0x10001;
+const VERSION: u32 = 1;
+
+pub struct ROSDriver<'a, T: Time> {
```

Comment:
```suggestion
pub struct ReadOnlySyscallDriver<'a, T: Time> {
```

This is a partially symptom of a different project I'm working on, but I can't think of anything but [robots](https://www.ros.org/) when I see ROS. I think it's worth the characters / clarity to just spell this out everywhere.